src/queueUntilType.js

```diff
@@ -0,0 +1,35 @@
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+export function queueUntilType(expectedType) {
+  return this.lift(function(source) {
+    let queue = [];
+    let hasSeenType = false;
+    const nullifyQueue = () => { queue = null; };
+
+    const outer = {
+      notifyNext: (_, action) => this.next(action),
+      notifyComplete: nullifyQueue
+    };
+    return source.subscribe({
+      next: action => {
+        // It's important to check this first because of possible re-entry
+        // before we've finished flushing the queue.
+        if (hasSeenType) {
+          return this.next(action);
+        }
+
+        if (action && action.type === expectedType) {
+          hasSeenType = true;
+          this.add(subscribeToResult(outer, queue));
+          // no longer needed, free up memory
+          nullifyQueue();
+        } else {
+          queue.push(action);
+        }
+      },
+      error: e => this.error(e),
+      complete: () => this.complete(),
+      unsubscribe: nullifyQueue
```

Comment:
> AFAIK passing unsubscribe isn't a supported thing for subscribers

As in this is something that should not be done?

From tracing the source code I can see it the observer is eventually wrapped in a SafeSubscriber, which will parse out the unsubscribe method correctly. http://reactivex.io/rxjs/file/es6/Subscriber.js.html#lineNumber160

> presumably the containing closure would be collected

I can remove it in this case.